### PR TITLE
add variable to define client id

### PR DESCRIPTION
**Pull request by phozzy**
intead of using hardcoded transmart-client as client_id we can add a special variable.

### CODE REVIEW

**Review by jochemb — CHANGES_REQUESTED** [review]
If there are sensible reasons to have different client ids for different instances, I'd like to learn them. In that case makes sense to have have a parameter for this. In any case I think we should either have a sensible default or make the client id discoverable (e.g. by exposing details via a public part of transmarts api). I'd prefer to not bother users of this library with connection details they do not know about.  

Anyways, I think it is fine to add this parameterisation, as long as you bring back the defaults.

File: transmart/api/auth.py [hunk 1]
```diff
@@ -6,19 +6,15 @@
...
         self.user = username or input('Username: ')
         self.password = password
         self.realm = realm
+        self.client_id = client_id
         self._access_token = None
         self.get_token()
 
-    @property
-    @abc.abstractmethod
-    def client_id(self):
```
Comment (jochemb) [inline transmart/api/auth.py:19]: could change this to `_default_client_id` and update the initialiser to use that if the client_id is None, eg.: `self.client_id = client_id or self._default_client_id`

Comment (phozzy) [reply transmart/api/auth.py:19]: @jochemb I have changed, take a look.

Comment (jochemb) [reply transmart/api/auth.py:19]: lgtm

**Review by jochemb — APPROVED** [review]

**Pull request merged**